src/Microsoft.DocAsCode.MarkdownLite/JsonRenderer.cs

```diff
@@ -0,0 +1,305 @@
+        {
+            return this.Insert("InLineCode", this.GetSize(), Type.Size);
+        }
+
+        public virtual StringBuffer Render(IMarkdownRenderer renderer, GfmDelInlineToken token, MarkdownInlineContext context)
+        {
+            var childContent = StringBuffer.Empty;
+            foreach (var item in token.Content)
+            {
+                childContent += renderer.Render(item);
+            }
+            return this.Insert("Del", childContent.ToString(), Type.Child);
+        }
+
+        public virtual StringBuffer Render(IMarkdownRenderer renderer, MarkdownTagInlineToken token, MarkdownInlineContext context)
+        {
+            return this.Insert("Tag", this.GetSize(), Type.Size);
+        }
+
+        public virtual StringBuffer Render(IMarkdownRenderer renderer, MarkdownBrInlineToken token, MarkdownInlineContext context)
+        {
+            return this.Insert("Br", this.GetSize(), Type.Size);
+        }
+
+        public virtual StringBuffer Render(IMarkdownRenderer renderer, MarkdownTextToken token, MarkdownInlineContext context)
+        {
+            return this.Insert("InLineText(" + token.Content.Replace("\n", "\\n") + ")", this.GetSize(), Type.Size);
+        }
+
+        #endregion
+
+        #region Misc
+
+        public virtual StringBuffer Render(IMarkdownRenderer renderer, MarkdownIgnoreToken token, IMarkdownContext context)
+        {
+            return this.Insert("Ignore", this.GetSize(), Type.Size);
+        }
+
+        public virtual StringBuffer Render(IMarkdownRenderer renderer, MarkdownRawToken token, IMarkdownContext context)
+        {
+            switch (token.Rule.Name)
+            {
+                case ("Html"):
+                {
+                    return this.Insert("Raw(FromHtml)", this.GetSize(), Type.Size);
+                }
+                case ("Inline.AutoLink"):
+                {
+                    return this.Insert("Raw(FromInLineAutoLink)", this.GetSize(), Type.Size);
+                }
+                case ("Inline.CodeElement"):
+                {
+                    return this.Insert("Raw(FromInLineCodeElement)", this.GetSize(), Type.Size);
+                }
+                case ("Inline.Comment"):
+                {
+                    return this.Insert("Raw(FromInLineComment)", this.GetSize(), Type.Size);
+                }
+                case ("GfmHtmlComment"):
+                {
+                    return this.Insert("Raw(FromGfmHtmlComment)", this.GetSize(), Type.Size);
+                }
+                case ("Inline.Gfm.Url"):
+                {
+                    return this.Insert("Raw(FromGfmUrl)", this.GetSize(), Type.Size);
+                }
+                default:
+                    return this.Insert("Raw", this.GetSize(), Type.Size);
+            }
+        }
+
+        #endregion
+
+        #region Private
+
+        private StringBuffer Insert(string name, string tokenContent, Type tokenType)
```

Comment:
agree ,name can be StringBuffer but tokenContent should be string for that class StringBuffer doesn't have 'remove' function

---
In reply to: [74728520](https://github.com/dotnet/docfx/pull/608#discussion_r74728520) [](ancestors = 74728520)